modules/ui/fields/localized.js

```diff
@@ -521,16 +521,52 @@ export function uiFieldLocalized(field, context) {
         loadCountryCode();
         return localized;
     };
+async function fetchCountryData() {
+    const response = await fetch('https://restcountries.com/v3.1/all');
+    const countries = await response.json();
+    const countryToISO = {};
+
+    countries.forEach(country => {
+        const countryName = country.name.common.toLowerCase();
+        const isoCode = country.cca2.toLowerCase();
+        countryToISO[countryName] = isoCode;
+    });
+
+    return countryToISO;
+}
+
+let countryToISO = {};
+
+async function getISOCode(input) {
+    if (Object.keys(countryToISO).length === 0) {
+        countryToISO = await fetchCountryData();
+    }
+
+    const lowerInput = input.toLowerCase();
 
-    function loadCountryCode() {
-        var extent = combinedEntityExtent();
-        var countryCode = extent && countryCoder.iso1A2Code(extent.center());
-        _countryCode = countryCode && countryCode.toLowerCase();
+    if (Object.values(countryToISO).includes(lowerInput)) {
+        return lowerInput;
     }
 
-    function combinedEntityExtent() {
-        return _entityIDs && _entityIDs.length && utilTotalExtent(_entityIDs, context.graph());
+    const isoCode = countryToISO[lowerInput];
+    if (isoCode) {
+        return isoCode;
+    } else {
+        return "ISO code or country name does not exist.";
     }
+}
+
+async function loadCountryCode() {
+    var extent = combinedEntityExtent();
+    var countryCode = extent && countryCoder.iso1A2Code(extent.center());
+    _countryCode = await getISOCode(countryCode) || (countryCode && countryCode.toLowerCase());
```

Comment:
As a reminder, the task is to display a list of countries by name and apply a country code in response to the user selecting a country name from the list. [`Intl.DisplayNames`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DisplayNames) is a useful API for converting country codes to names in the user’s language (only for display purposes).

Reply:
import { countryCoder } from '@rapideditor/country-coder';
// ... (other existing imports)

export function uiFieldCombo(field, context) {
    // ... (existing code)

    let _countryList;

    function getCountryList() {
        if (!_countryList) {
            const allCountries = countryCoder.countryList;
            _countryList = allCountries.map(country => ({
                value: country.code.toLowerCase(),
                label: new Intl.DisplayNames([context.locale()], { type: 'region' }).of(country.code)
            }));
            _countryList.sort((a, b) => a.label.localeCompare(b.label, context.locale()));
        }
        return _countryList;
    }

    function setStaticValues(callback, filter) {
        if (field.key === 'country') {
            _comboData = getCountryList();
        } else {
            _comboData = getOptions();
        }

        if (filter) {
            _comboData = _comboData.filter(filter);
        }

        _comboData = objectDifference(_comboData, _multiData);
        _combobox.data(_comboData);
        if (callback) callback(_comboData);
    }

    function combo(selection) {
        // ... (existing code)

        if (field.key === 'country') {
            _combobox
                .fetcher(function(value, callback) {
                    const filtered = getCountryList().filter(d => 
                        d.label.toLowerCase().includes(value.toLowerCase()) || 
                        d.value.includes(value.toLowerCase())
                    );
                    callback(filtered);
                });
        } else if (_showTagInfoSuggestions && services.taginfo) {
            selection.call(_combobox.fetcher(setTaginfoValues), attachTo);
            setTaginfoValues('', setPlaceholder);
        } else {
            selection.call(_combobox, attachTo);
            setStaticValues(setPlaceholder);
        }

        // ... (rest of existing code)
    }

    // ... (rest of existing code)
}